domain/src/main/kotlin/no/nav/su/se/bakover/domain/behandlinger/stopp/StoppbehandlingService.kt

```diff
@@ -0,0 +1,103 @@
+package no.nav.su.se.bakover.domain.behandlinger.stopp
+
+import arrow.core.Either
+import arrow.core.right
+import no.nav.su.se.bakover.common.UUIDFactory
+import no.nav.su.se.bakover.common.idag
+import no.nav.su.se.bakover.common.now
+import no.nav.su.se.bakover.domain.Attestant
+import no.nav.su.se.bakover.domain.Sak
+import no.nav.su.se.bakover.domain.Saksbehandler
+import no.nav.su.se.bakover.domain.behandlinger.stopp.StoppbehandlingService.ValidertStoppbehandling.Companion.validerStoppbehandling
+import no.nav.su.se.bakover.domain.beregning.BeregningsPeriode
+import no.nav.su.se.bakover.domain.oppdrag.NyUtbetaling
+import no.nav.su.se.bakover.domain.oppdrag.Oppdrag
+import no.nav.su.se.bakover.domain.oppdrag.simulering.SimuleringClient
+import no.nav.su.se.bakover.domain.oppdrag.simulering.SimuleringFeilet
+import java.time.Clock
+import java.time.LocalDate
+import java.time.temporal.TemporalAdjusters
+
+class StoppbehandlingService(
+    private val simuleringClient: SimuleringClient,
+    private val clock: Clock = Clock.systemUTC(),
+    private val uuidFactory: UUIDFactory = UUIDFactory(),
+    private val stoppbehandlingRepo: StoppbehandlingRepo
+) {
+    object KunneIkkeOppretteStoppbehandling
+
+    fun stoppUtbetalinger(
+        sak: Sak,
+        saksbehandler: Saksbehandler,
+        stoppÅrsak: String
+    ): Either<KunneIkkeOppretteStoppbehandling, Stoppbehandling> {
+        return stoppbehandlingRepo.hentPågåendeStoppbehandling(sak.id)?.right()
+            ?: nyStoppbehandling(sak, saksbehandler, stoppÅrsak)
+                .mapLeft { KunneIkkeOppretteStoppbehandling }
+                .map { stoppbehandlingRepo.opprettStoppbehandling(it) }
+    }
+
+    private fun nyStoppbehandling(
+        sak: Sak,
+        saksbehandler: Saksbehandler,
+        stoppÅrsak: String,
+    ): Either<SimuleringFeilet, Stoppbehandling.Simulert> {
+
+        val stoppesFraOgMed = idag(clock).with(TemporalAdjusters.firstDayOfNextMonth())
+        val validertStoppbehandling = sak.oppdrag.validerStoppbehandling(stoppesFraOgMed)
```

Comment:
Tanker rundt utbetalinger som er sendt men mangler kvittering?

Reply:
Jeg tenker det er litt vrient :) Det letteste for oss er å ikke tillate det foreløpig.